libcxx/include/generator

```diff
@@ -0,0 +1,593 @@
+// -*- C++ -*-
```

Comment:
Does this PR also address

- https://cplusplus.github.io/LWG/issue3762 generator::iterator::operator== should pass by reference
- https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2023/p2787r1.pdf pmr::generator - Promise Types are not Values
- https://cplusplus.github.io/LWG/issue3894 generator::promise_type::yield_value(ranges::elements_of<Rng, Alloc>) should not be noexcept

These seem small and I think they can be done in this PR.